x-pack/plugins/lens/public/types.ts

```diff
@@ -66,6 +66,7 @@ export interface Datasource<T = unknown, P = unknown> {
 export interface DatasourcePublicAPI {
   getTableSpec: () => TableSpec;
   getOperationForColumnId: (columnId: string) => Operation | null;
+  generateColumnId: () => string;
```

Comment:
I would appreciate it if the function order were consistent- I think `generateColumnId` has different ordering in different parts of this PR.